packages/pds/src/db/migrations/20230124T230923517Z-user-partitioned-cids.ts

```diff
@@ -0,0 +1,84 @@
+import { Kysely, sql } from 'kysely'
+import { Dialect } from '..'
+
+export async function up(db: Kysely<any>, dialect: Dialect): Promise<void> {
+  const binaryDatatype = dialect === 'sqlite' ? 'blob' : sql`bytea`
+
+  await db.schema
+    .createTable('ipld_block_temp')
+    .addColumn('cid', 'varchar', (col) => col.notNull())
+    .addColumn('creator', 'varchar', (col) => col.notNull())
+    .addColumn('size', 'integer', (col) => col.notNull())
+    .addColumn('content', binaryDatatype, (col) => col.notNull())
+    .addColumn('indexedAt', 'varchar', (col) => col.notNull())
+    .addPrimaryKeyConstraint('ipld_block_with_creator_pkey', ['cid', 'creator'])
+    .execute()
+
+  await db
+    .insertInto('ipld_block_temp')
+    .expression((exp) =>
+      exp
+        .selectFrom('ipld_block')
+        .innerJoin(
+          'ipld_block_creator',
+          'ipld_block_creator.cid',
+          'ipld_block.cid',
+        )
+        .select([
+          'ipld_block.cid as cid',
+          'ipld_block_creator.did as creator',
+          'ipld_block.size as size',
+          'ipld_block.content as content',
+          'ipld_block.indexedAt as indexedAt',
+        ]),
```

Comment:
Just something to be a little careful with here— I don't think the column aliases are doing what one might expect/hope them to do.  I believe all that matters is the column order.  To ensure these are lining-up, might consider adding `.columns(...)` to the query (e.g. https://koskimas.github.io/kysely/classes/InsertQueryBuilder.html#expression).

Reply:
this is actually in regards to a separate kysely thing that bit me once.

on `.returning()`
> Note that on SQLite you need to give aliases for the expressions to avoid [this bug](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) in SQLite. For example .returning('id as id').

altho i just kinda had it in my head that i _always_ had to supply an alias, which it turns out you don't for `select()`

Reply:
but i'll go ahead & add `.columns()` as well to be safe 👌

Reply:
Oh wild, good to know!